core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -108,19 +109,26 @@ func (ec *ethConfirmer) CheckForReceipts() error {
 	for _, etx := range unconfirmedEtxs {
 		for _, attempt := range etx.EthTxAttempts {
 			// NOTE: If this becomes a performance bottleneck due to eth node requests,
-			// it may be possible to use goroutines here to speed it up
+			// it may be possible to use goroutines here to speed it up by
+			// issuing `fetchReceipt` requests in parallel
 			receipt, err := ec.fetchReceipt(attempt.Hash)
-			if err != nil {
-				return err
+			if isParityQueriedReceiptTooEarly(err) {
+				logger.Debugf("EthConfirmer: got receipt for transaction %s but it's still in the mempool and not included in a block yet", attempt.Hash.Hex())
+				break
```

Comment:
Probably shouldn't break here, as we'll fail to check the rest of the attempts.

Reply:
Do we have to check all the attempts for each transaction?
Is there no way to filter them?
_#noobquestions_

Reply:
Having slept on this, I actually think `break`ing is the correct behaviour.

A node can only have one transaction on-chain or in mempool with a particular nonce. Thus, if we get this parity receipt for an attempt, none of the other attempts will have a receipt (unless the node is switched out on us for one with a different worldview mid-loop, which seems like a tiny edge case which would be rectified on the next head anyway).

It is therefore safe and sensible to break here.

Reply:
Thanks for the explanation. Would it make sense to add this as a comment before that `break`?